SwiftBasicsExercise.playground/Pages/ControlFlow_5.1.xcplaygroundpage/Contents.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+//Control-flow exercise 1
+print("\n1-->")
+for i in stride(from: 0, to: 100,by: 1) {
+    let power = Int(pow(Float(2), Float(i)))
+    if power > 100 {
+        break
+    } else {
+        print("\(power) ", terminator: "")
+    }
+}
+
+//Control-flow exercise 2
+print("\n\n2-->")
+var n = 3
+for _ in 1...n {
+    for _ in 1...n {
+        print("*", terminator: "")
+    }
+    print()
+}
+
+//Control-flow exercise 3
+print("\n3-->")
+for i in 1...5 {
+    for _ in 1...i {
+        print("* ", terminator: "")
+    }
+    print()
+}
+
+//Control-flow exercise 4
+print("\n4-->")
+for i in 1...5 {
+    for _ in 0..<5-i {
+            print(" ", terminator: "")
+    }
+    for _ in 0..<i {
+        print("*", terminator: "")
+    }
+    print()
+}
```

Comment:
this can be simplified too
let N = 4
(1...N).forEach { i in
let leftPadding = String(repeating: " ", count: N-i)
print(leftPadding + String(repeating: "*", count: i))
}